huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go

```diff
@@ -307,100 +359,124 @@ func resourceDmsKafkaInstanceCreate(ctx context.Context, d *schema.ResourceData,
 		createOpts.Tags = taglist
 	}
 
-	logp.Printf("[DEBUG] Create Options: %#v", createOpts)
+	logp.Printf("[DEBUG] Create DMS Kafka instance options: %#v", createOpts)
 	// Add password here so it wouldn't go in the above log entry
 	createOpts.Password = d.Get("password").(string)
 	createOpts.KafkaManagerPassword = d.Get("manager_password").(string)
 
 	v, err := instances.Create(dmsV2Client, createOpts).Extract()
 	if err != nil {
-		return fmtp.DiagErrorf("error creating HuaweiCloud dms kafka instance: %s", err)
+		return fmtp.DiagErrorf("error creating HuaweiCloud DMS kafka instance: %s", err)
 	}
 	logp.Printf("[INFO] instance ID: %s", v.InstanceID)
 
+	// Store the instance ID now
+	d.SetId(v.InstanceID)
+
 	stateConf := &resource.StateChangeConf{
 		Pending:      []string{"CREATING"},
 		Target:       []string{"RUNNING"},
 		Refresh:      DmsKafkaInstanceStateRefreshFunc(dmsV2Client, v.InstanceID),
 		Timeout:      d.Timeout(schema.TimeoutCreate),
 		Delay:        300 * time.Second,
-		MinTimeout:   3 * time.Second,
-		PollInterval: 10 * time.Second,
+		PollInterval: 15 * time.Second,
 	}
 	_, err = stateConf.WaitForStateContext(ctx)
 	if err != nil {
 		return fmtp.DiagErrorf("error waiting for instance (%s) to become ready: %s", v.InstanceID, err)
 	}
 
-	// Store the instance ID now
-	d.SetId(v.InstanceID)
+	// resize storage capacity of the instance
+	storage_space, ok := d.GetOk("storage_space")
```

Comment:
better to check whether storage_space less than default storage_space before creating the instance.